pkg/medusa/reconcile.go

```diff
@@ -360,3 +361,16 @@ func addOrUpdateAdditionalVolume(dcConfig *cassandra.DatacenterConfig, volume *v
 		dcConfig.StorageConfig.AdditionalVolumes[volumeIndex] = *volume
 	}
 }
+
+// If auth is enabled in this cluster, we need to allow Medusa to access the cluster through CQL. This is done by
+// declaring a Cassandra user whose credentials are pulled from CassandraUserSecretRef.
+func AddCqlUser(medusaTemplate *api.MedusaClusterTemplate, dcConfig *cassandra.DatacenterConfig, clusterName string) {
```

Comment:
The `reaper` package defines a nearly identical function (see [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/pkg/reaper/datacenter.go#L33)). The only difference is that the first argument is a `*reaperapi.ReaperClusterTemplate`. Could you refactor these into a common, shared function?

Reply:
Good idea 👍 

Done!